Dockerfile-linux.template

```diff
@@ -0,0 +1,201 @@
+{{ ) end -}}
+	url=; \
+	case "$arch" in \
+{{
+	[
+		.arches | to_entries[]
+		| select(.value.supported)
+		| .key as $bashbrewArch
+		| (
+			os_arches
+			| .[$bashbrewArch] // empty
+		) as $osArch
+		| .value
+		| (
+-}}
+		{{ $osArch | @sh }}) \
+{{ if .url and (is_alpine | not) then ( -}}
+			url={{ .url | @sh }}; \
+			sha256={{ .sha256 | @sh }}; \
+{{ ) else ( -}}
+			export {{ .env | to_entries | map(.key + "=" + (.value | @sh)) | join(" ") }}; \
+{{ ) end -}}
+			;; \
+{{
+		)
+	] | add
+-}}
+		*) echo >&2 "error: unsupported architecture '$arch' (likely packaging update needed)"; exit 1 ;; \
+	esac; \
+	build=; \
+	if [ -z "$url" ]; then \
+# https://github.com/golang/go/issues/38536#issuecomment-616897960
+		build=1; \
+		url={{ .arches.src.url | @sh }}; \
+		sha256={{ .arches.src.sha256 | @sh }}; \
+{{ if is_alpine then ( -}}
+# the precompiled binaries published by Go upstream are not compatible with Alpine, so we always build from source here 😅
+{{ ) else ( -}}
+		echo >&2; \
+		echo >&2 "warning: current architecture ($arch) does not have a compatible Go binary release; will be building from source"; \
+		echo >&2; \
+{{ ) end -}}
+	fi; \
+	\
+	wget -O go.tgz.asc "$url.asc"; \
+	wget -O go.tgz "$url"{{ if is_alpine then "" else " --progress=dot:giga" end }}; \
+	echo "$sha256 *go.tgz" | sha256sum -c -; \
+	\
+# https://github.com/golang/go/issues/14739#issuecomment-324767697
+	GNUPGHOME="$(mktemp -d)"; export GNUPGHOME; \
+# https://www.google.com/linuxrepositories/
+	gpg --batch --keyserver keyserver.ubuntu.com --recv-keys 'EB4C 1BFD 4F04 2F6D DDCC EC91 7721 F63B D38B 4796'; \
+	gpg --batch --verify go.tgz.asc go.tgz; \
+	gpgconf --kill all; \
+	rm -rf "$GNUPGHOME" go.tgz.asc; \
+	\
+	tar -C /usr/local -xzf go.tgz; \
+	rm go.tgz; \
+	\
+	if [ -n "$build" ]; then \
+{{ if is_alpine then ( -}}
+		apk add --no-cache --virtual .build-deps \
+			bash \
+			gcc \
+			go \
+			musl-dev \
+		; \
+{{ ) else ( -}}
+		savedAptMark="$(apt-mark showmanual)"; \
+		apt-get update; \
+		apt-get install -y --no-install-recommends golang-go; \
+{{ ) end -}}
+		\
+		( \
+			cd /usr/local/go/src; \
+# set GOROOT_BOOTSTRAP + GOHOST* such that we can build Go successfully
+			export GOROOT_BOOTSTRAP="$(go env GOROOT)" GOHOSTOS="$GOOS" GOHOSTARCH="$GOARCH"; \
+{{
+	def need_386_hack:
+		is_alpine and alpine_version == "3.13"
+-}}
+{{ if need_386_hack then ( -}}
+			if [ -n "${GO386:-}" ]; then \
+# https://github.com/docker-library/golang/issues/359 -> https://github.com/golang/go/issues/44500
+# Go 1.16 + Alpine 3.13 == Go 1.15 bootstrap
+# (once Alpine 3.13 goes away, we can remove this)
+				GO386= ./bootstrap.bash; \
+				export GOROOT_BOOTSTRAP="/usr/local/go-$GOOS-$GOARCH-bootstrap"; \
+				"$GOROOT_BOOTSTRAP/bin/go" version; \
+			fi; \
+{{ ) else "" end -}}
+			./make.bash; \
+{{ if need_386_hack then ( -}}
+			if [ -n "${GO386:-}" ]; then \
+				rm -rf "$GOROOT_BOOTSTRAP"; \
+			fi; \
+{{ ) else "" end -}}
```

Comment:
This looks silly to make a function to use twice to avoid having an else that duplicates the `./make.bash` line. (It's not new to this PR, so we can fix it later).